cmd/explaintest/r/tpch.result

```diff
@@ -1088,14 +1088,14 @@ and l_shipinstruct = 'DELIVER IN PERSON'
-  └─IndexMergeJoin_43	6286493.79	root		inner join, inner:TableReader_38, outer key:tpch.lineitem.l_partkey, inner key:tpch.part.p_partkey, other cond:or(and(and(eq(tpch.part.p_brand, "Brand#52"), in(tpch.part.p_container, "SM CASE", "SM BOX", "SM PACK", "SM PKG")), and(ge(tpch.lineitem.l_quantity, 4), and(le(tpch.lineitem.l_quantity, 14), le(tpch.part.p_size, 5)))), or(and(and(eq(tpch.part.p_brand, "Brand#11"), in(tpch.part.p_container, "MED BAG", "MED BOX", "MED PKG", "MED PACK")), and(ge(tpch.lineitem.l_quantity, 18), and(le(tpch.lineitem.l_quantity, 28), le(tpch.part.p_size, 10)))), and(and(eq(tpch.part.p_brand, "Brand#51"), in(tpch.part.p_container, "LG CASE", "LG BOX", "LG PACK", "LG PKG")), and(ge(tpch.lineitem.l_quantity, 29), and(le(tpch.lineitem.l_quantity, 39), le(tpch.part.p_size, 15))))))
-    ├─TableReader_29(Build)	6286493.79	root		data:Selection_28
-    │ └─Selection_28	6286493.79	cop[tikv]		eq(tpch.lineitem.l_shipinstruct, "DELIVER IN PERSON"), in(tpch.lineitem.l_shipmode, "AIR", "AIR REG"), or(and(ge(tpch.lineitem.l_quantity, 4), le(tpch.lineitem.l_quantity, 14)), or(and(ge(tpch.lineitem.l_quantity, 18), le(tpch.lineitem.l_quantity, 28)), and(ge(tpch.lineitem.l_quantity, 29), le(tpch.lineitem.l_quantity, 39))))
-    │   └─TableFullScan_27	300005811.00	cop[tikv]	table:lineitem	keep order:false
-    └─TableReader_38(Probe)	0.80	root		data:Selection_37
-      └─Selection_37	0.80	cop[tikv]		ge(tpch.part.p_size, 1), or(and(eq(tpch.part.p_brand, "Brand#52"), and(in(tpch.part.p_container, "SM CASE", "SM BOX", "SM PACK", "SM PKG"), le(tpch.part.p_size, 5))), or(and(eq(tpch.part.p_brand, "Brand#11"), and(in(tpch.part.p_container, "MED BAG", "MED BOX", "MED PKG", "MED PACK"), le(tpch.part.p_size, 10))), and(eq(tpch.part.p_brand, "Brand#51"), and(in(tpch.part.p_container, "LG CASE", "LG BOX", "LG PACK", "LG PKG"), le(tpch.part.p_size, 15)))))
-        └─TableRangeScan_36	1.00	cop[tikv]	table:part	range: decided by [tpch.lineitem.l_partkey], keep order:true
+└─Projection_46	733887.82	root		mul(tpch.lineitem.l_extendedprice, minus(1, tpch.lineitem.l_discount))->Column#28
+  └─HashJoin_45	733887.82	root		inner join, equal:[eq(tpch.part.p_partkey, tpch.lineitem.l_partkey)], other cond:or(and(and(eq(tpch.part.p_brand, "Brand#52"), in(tpch.part.p_container, "SM CASE", "SM BOX", "SM PACK", "SM PKG")), and(ge(tpch.lineitem.l_quantity, 4), and(le(tpch.lineitem.l_quantity, 14), le(tpch.part.p_size, 5)))), or(and(and(eq(tpch.part.p_brand, "Brand#11"), in(tpch.part.p_container, "MED BAG", "MED BOX", "MED PKG", "MED PACK")), and(ge(tpch.lineitem.l_quantity, 18), and(le(tpch.lineitem.l_quantity, 28), le(tpch.part.p_size, 10)))), and(and(eq(tpch.part.p_brand, "Brand#51"), in(tpch.part.p_container, "LG CASE", "LG BOX", "LG PACK", "LG PKG")), and(ge(tpch.lineitem.l_quantity, 29), and(le(tpch.lineitem.l_quantity, 39), le(tpch.part.p_size, 15))))))
+    ├─TableReader_32(Build)	24323.12	root		data:Selection_31
```

Comment:
I used a 50M data for testing. It turns out the new estimated row count is closer to the actual row count. 

Originally it uses a default 0.8 as the selectivity on the `Selection` on the  `part` table and the est. row count after the selection is 8000.00, now it can calculate the selectivity which is close to actual selectivity and the est. row count is 24.75, and the actual row count is 21.

Originally the est. row count after join is 7025.23, now it's 741.93, actually it's 5.

Reply:
Verified on tpch 50g test. The new estimated row count is closer to the actual row count.

Originally it uses a default 0.8 as the selectivity of the Selection on the `part` table, now it can calculate the selectivity which is closer to the actual selectivity. Now the est. row count is ~24000, and the actual row count is 24150.

Originally the est. row count after join is 6,286,493, now it's ~700,000, actually it's 5,665.

Reply:
result of running `explain analyze` on query 19 (remove some fields for visibility):
```
id	estRows	actRows	task	access object
HashAgg_11	1.00	1	root
└─Projection_46	696411.91	5665	root
  └─HashJoin_26	696411.91	5665	root
    ├─TableReader_32(Build)	23423.35	24150	root
    │ └─Selection_31	23423.35	0	cop[tikv]
    │   └─TableFullScan_30	10000000.00	0	cop[tikv]	table:part
    └─TableReader_29(Probe)	6950210.29	7073612	root
      └─Selection_28	6950210.29	0	cop[tikv]
        └─TableFullScan_27	300005811.00	0	cop[tikv]	table:lineitem
```

Reply:
and the result of running `explain analyze` on query 19 on the master branch:
```
id	estRows	actRows	task	access object
HashAgg_11	1.00	1	root
└─Projection_46	6950210.29	5665	root
  └─HashJoin_25	6950210.29	5665	root
    ├─TableReader_32(Build)	8000000.00	24150	root
    │ └─Selection_31	8000000.00	0	cop[tikv]
    │   └─TableFullScan_30	10000000.00	0	cop[tikv]	table:part
    └─TableReader_29(Probe)	6950210.29	7073612	root
      └─Selection_28	6950210.29	0	cop[tikv]
        └─TableFullScan_27	300005811.00	0	cop[tikv]	table:lineitem
```